terraform/variables.go

```diff
@@ -9,15 +10,38 @@ import (
 
 type ValueMap map[string]cty.Value
 
+type ExpressionMap map[string]hcl.Expression
+
 type ModuleVariables struct {
 	inputs ValueMap
 	locals ValueMap
 }
 
-type ParserVariables map[string]ValueMap
-
 type InputVariablesByFile map[string]ValueMap
 
+// ExtractVariables extracts the input variables and local values from the provided file
+func ExtractVariables(file File) (ValueMap, ExpressionMap, error) {
+	inputsMap := ValueMap{}
+	fileLocals := ExpressionMap{}
+	var diagnostics hcl.Diagnostics
+
+	if isValidInputVariablesFile(file.fileName) {
+		inputsMap, diagnostics = extractInputVariablesFromFile(file)
+		if diagnostics.HasErrors() {
```

Comment:
The only way we'd get errors here is if the Terraform file was in any way invalid. I don't think it's worth trying to extract the locals after because we'd just get the same error again, but before we would do that. Open to discussions